src/services/item/plugins/invitation/index.ts

```diff
@@ -192,6 +210,59 @@ const plugin: FastifyPluginAsync = async (fastify) => {
       reply.status(StatusCodes.NO_CONTENT);
     },
   );
+
+  fastify.post<{ Params: { itemId: string } }>(
+    '/:itemId/enroll',
+    {
+      schema: enroll,
+      preHandler: [isAuthenticated, matchOne(validatedMemberAccountRole)],
+    },
+    async ({ user, params }) => {
+      const member = notUndefined(user?.account);
+      assertIsMember(member);
+      const { itemId } = params;
+
+      return await db.transaction(async (manager) => {
+        const repositories = buildRepositories(manager);
+
+        const item = await itemService.get(
+          member,
+          repositories,
+          itemId,
+          PermissionLevel.Read,
+          false,
+        );
+
+        if (!(await itemLoginService.getByItemPath(repositories, item.path))) {
+          const error = new ItemLoginSchemaNotFound();
+          error.statusCode = StatusCodes.FORBIDDEN;
+          throw error;
+        }
+
+        // Check if the member already has a membership, if so, throw an error
+        let hasPermission = true;
+        try {
+          await validatePermission(repositories, PermissionLevel.Read, member, item);
+        } catch (err) {
+          hasPermission = err.statusCode !== StatusCodes.FORBIDDEN;
+        }
+        if (hasPermission) {
+          throw new ItemMembershipAlreadyExists();
+        }
+
+        return await itemMembershipService.post(
```

Comment:
I hope one day we will rename the methods of the services ^^ `post` --> `create`